File: Data Structures & Algorithms/first-missing-positive/submission-4.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int i = 0;

        while (i < nums.length) {
            if (nums[i] <= 0 || nums[i] >= nums.length) {
                i++;
                continue;
            }
            int index = nums[i] - 1;
            if (nums[i] != nums[index]) {
                int swap = nums[index];
                nums[index] = nums[i];
                nums[i] = swap;
            } else {
                i++;
            }
        }

        for (int j = 0; j < nums.length; j++) {
            if (nums[j] != j + 1) {
                return j + 1;
            }
        }

        return nums.length + 1;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        val set = mutableSetOf<Int>()
        for (num in nums) {
            set.add(num)
        }

        val starts = mutableListOf<Int>()
        var maxLength = 0
        for (num in nums) {
            if (!set.contains(num - 1)) {
                var sequenceElement = num
                var length = 0

                while (set.contains(sequenceElement)) {
                    set.remove(sequenceElement)
                    sequenceElement++
                    length++
                }

                maxLength = max(length, maxLength)
            }
        }

        return maxLength
    }
}
